### ISSUE TITLE
Ignore doesn't ignore while copying files to temp dir

### ISSUE THREAD

**Issue by anilanar** [ojkelly/yarn.build#215]
**Describe the bug**
When there are files that have permissions incompatible with whatever yarn bundle does with them (or yarn/fslib does with them), adding such files to .bundleignore does not help because apparently ignoring happens later on in the pipeline.

In my case, I'm using nix + direnv, which creates a `.direnv` directory that contains symlinks to readonly files somewhere else on the system. Even if I add `.direnv` to `.bundleignore`, `yarn bundle` copies them to a temp dir and then runs `chmod` on them, which fails.

**To Reproduce**

1. cd into a project that uses yarn v2/v3 workspaces
1. `touch ~/bar` need a file somewhere else for this reproduction
1. `sudo chown root ~/bar` root or any other user, any user other than yours.
1. `sudo chmod 0444 ~/bar` make it readonly for everybody.
1. `ln -s ~/bar foo` gotta create a symlink for reproduction
1. Add foo to `.bundleignore` 
1. cd into a workspace and run `yarn bundle`.

I think this can also be reproduced by creating a symlink to a non-existing path.

**Expected behavior**
Everything works fine.

**Actual behavior**
`EPERM: operation not permitted, chmod '<omitted tmp dir path>/foo'`

**Desktop (please complete the following information):**
 - MacOS

**Comment by ojkelly** [comment]
Yep I think you're correct it's happening too late in the process.

The current process they're deleted after being copied to a tmp directory.

https://github.com/ojkelly/yarn.build/blob/2f000664c0255a0bd63e1180e679c5ede2e63fac/packages/plugins/plugin-bundle/src/commands/bundle/index.ts#L330

This can probably be solved by using something like `fs-extra` which has the ability to take in a function to filter files to copy
https://github.com/jprichardson/node-fs-extra/blob/master/docs/copy.md#copysrc-dest-options-callback

it would be replacing this line https://github.com/ojkelly/yarn.build/blob/2f000664c0255a0bd63e1180e679c5ede2e63fac/packages/plugins/plugin-bundle/src/commands/bundle/index.ts#L286

If this feels like something you might be able to work on let me know.
